CHANGELOG.md

```diff
@@ -7,6 +7,32 @@ and this project's packages adheres to [Semantic Versioning](http://semver.org/s
 
 ## [Unreleased]
 
+### Changed
+
+- Upgrade to ingress-nginx [v0.40.1](https://github.com/kubernetes/ingress-nginx/blob/master/Changelog.md#0401).
+
+  **Important** upstream changes to pay special attention to:
+
+  - App/chart requires Kubernetes 1.16+ based platform release
+    - Note to maintainers: if NGINX IC App fix has to be done for Kubernetes 1.15 based platform releases, one of the previous releases has to be used as base)
+  - **Breaking change**: Validating `extensions/v1beta1` Ingress resources is no longer supported (to be able to support `networking.k8s.io/v1beta1` and Kubernetes 1.19+ `networking.k8s.io/v1` Ingress APIs), validating webhook which is enabled by default will reject their creation and any changes
+
+    Migration options:
+
+    - (*prefered*) Before upgrading to this or newer NGINX IC App, ensure that all existing and new Ingress resources use `networking.k8s.io/v1beta1` API (available since Kubernetes 1.14) or even newer `networking.k8s.io/v1` API (available sicne Kubernetes 1.19)
+    - Before or as part of upgrading to this or newer NGINX IC App, disable validating webhook (by setting `controller.admissionWebhooks.enabled` to `false`), and migrate to new Ingress APIs later, accepting risk that in the meantime invalid Ingress resources can be submitted and break NGINX IC configuration
+
+  - Default configuration changes:
+
+    - [`gzip-level`](https://github.com/kubernetes/ingress-nginx/blob/master/docs/user-guide/nginx-configuration/configmap.md#gzip-level) default changed from `5` to `1`
+    - [`ssl-session-tickets`](https://github.com/kubernetes/ingress-nginx/blob/master/docs/user-guide/nginx-configuration/configmap.md#ssl-session-tickets) default changed from `true` to `false`
+    - [`use-gzip`](https://github.com/kubernetes/ingress-nginx/blob/master/docs/user-guide/nginx-configuration/configmap.md#use-gzip) default changed from `true` to `false`
+    - [`upstream-keepalive-connections`](https://github.com/kubernetes/ingress-nginx/blob/master/docs/user-guide/nginx-configuration/configmap.md#upstream-keepalive-connections) changed from `32` to `320`
+    - [`upstream-keepalive-requests`](https://github.com/kubernetes/ingress-nginx/blob/master/docs/user-guide/nginx-configuration/configmap.md#upstream-keepalive-requests) changed from `100` to `10000`
+- Support and enable by default [mimalloc](https://github.com/microsoft/mimalloc) as a drop-in malloc replacement to reduce nginx memory utilization.
```

Comment:
These changes to default configuration improved performance in upstream performance tests https://github.com/kubernetes/ingress-nginx/pull/6226#issuecomment-698647569

//cc @whites11